style/pre-commit

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# A hook script to automatically fix code style and formatting errors with spotless
+mvn spotless:apply
```

Comment:
Prefer `.mvnw`

```suggestion
./mvnw spotless:apply
```

This is not a correct hook. A check hook should prevent the commit if the codes style conflict with the rules but it just fix the codestyle and let the commit succeeds. If the developers think the commit succeeds and then `git push`, the bad style codes are still pushed, leaving the fixed one locally. You can try modify some codes to bad style and try `git add . && git commit -m'Test'`, everything is going to succeed and if you then `git push`, you might don't know the hook script helped you fixed the codes and you might be curious why there are uncommitted codes.


Try mine here 👇

```suggestion
if ./mvnw spotless:check; then
  exit 0
fi

./mvnw spotless:apply

exit 1
```

Reply:
Oh thx for pointing this out, it's my first time to write a pre-commit hook. I will try your suggestions : )